examples/step-77/doc/intro.dox

```diff
@@ -0,0 +1,245 @@
+previously computed Jacobian matrix $J_{k-s}$ that goes back $s$
+steps, or whether we need to -- at least in this iteration -- actually
+re-compute the Jacobian $J_k$ and the corresponding preconditioner?
+This is, like the issue with line search, one that requires a
+non-trivial amount of code that monitors the convergence of the
+overall algorithm. We *could* implement these sorts of things
+ourselves, but we probably *shouldn't*: KINSOL already does that for
+us. It will tell our code when to "update" the Jacobian matrix.
+
+One last consideration if we were to use an iterative solver instead of
+the sparse direct one mentioned above: Not only is it possible to get
+away with replacing $J_k$ by some approximation $\tilde J_k$ when
+solving for the update $\delta U_k$, but one can also ask whether it
+is necessary to solve the linear system
+@f{align*}{
+  \tilde J_k \widetilde{\delta U}_k = -F_k
+@f}
+to high accuracy. The thinking goes like this: While our current solution
+$U_k$ is still far away from $U^\ast$, why would we solve this linear
+system particularly accurately? The update
+$U_{k+1}=U_k + \widetilde{\delta U}_k$ is likely still going to be far away
+from the exact solution, so why spend much time on solving the linear system
+to great accuracy? This is the kind of thinking that underlies algorithms
+such as the "Eisenstat-Walker trick" @cite eiwa96 in which one is given
+a tolerance to which the linear system above in iteration $k$ has to be
+solved, with this tolerance dependent on the progress in the overall
+nonlinear solver. As before, one could try to implement this oneself,
+but KINSOL already provides this kind of information for us -- though we
+will not use it in this program since we use a direct solver that requires
+no solver tolerance and just solves the linear system exactly up to
+round-off.
+
+As a summary of all of these considerations, we could say the
+following: There is no need to reinvent the wheel. Just like deal.II
+provides a vast amount of finite-element functionality, %SUNDIALS'
+KINSOL package provides a vast amount of nonlinear solver
+functionality, and we better use it.
+
+
+<h3> How deal.II interfaces with KINSOL </h3>
+
+KINSOL, like many similar packages, works in a pretty abstract way. At
+its core, it sees a nonlinear problem of the form
+@f{align*}{
+    F(U) = 0
+@f}
+and constructs a sequence of iterates $U_k$ which, in general, are
+vectors of the same length as the vector returned by the function
+$F$. To do this, there are a few things it needs from the user:
+- A way to resize a given vector to the correct size.
+- A way to evaluate, for a given vector $U$, the function $F(U)$. This
+  function is generally called the "residual" operation because the
+  goal is of course to find a point $U^\ast$ for which $F(U^\ast)=0$;
+  if $F(U)$ returns a nonzero vector, then this is the
+  "[residual](https://en.wikipedia.org/wiki/Residual_(numerical_analysis))"
```

Comment:
Is Mardown understood? In the PDF document it does not look right.

Reply:
doxygen does markdown, but apparently not if you put things in quotes...